#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QMessageBox>
#include <recordtablemodel.h>
#include <databasemanager.h>
#include <qprogressbar.h>
QT_BEGIN_NAMESPACE
namespace Ui { class MainWindow; }
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

	RecordTableModel* recordmodel;
    DatabaseManager dbmanager;
    QProgressBar* statusProgressBar;
    bool dbIsOpened =false;
public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();
public slots:
    void openFile();
    void createRecordDialog();
    void changeRecordDialog();
    void deleteRecordDialog();
    void createFile();

private:
    Ui::MainWindow *ui;
};
#endif // MAINWINDOW_H
